### PR TITLE
bug 1152386 - Remove sleep() calls from header tests.  Updating documentation.  Enable wiki tests.

### PR DESCRIPTION
**Pull request by darkwing**
The test update is a superficial one -- I just wanted both test files to be next to each other :)

As for the test, we know that `sleep()` is the equivalent to `setInterval` and we're essentially "hoping" the test is done.  To remove the sleep calls, we're using `executeAsync` to add a `transitionend` event, the same event we're using to move the box in our CSS.

The only odd point is it seems I need to call `click()` before adding the event, which doesn't seem good.  We can review that in the future but this update is a good step forward.

### CODE REVIEW

File: tests/ui/tests/header.js [hunk 1]
```diff
@@ -45,35 +45,54 @@ define([
...
             var homeSearchBoxId = 'home-q';
-            var originalSize;
+            var originalWidth;
 
-            return this.remote
+            remote.setExecuteAsyncTimeout(config.asyncExecutionTimeout);
+
+            var transitionEndCallback = function() {
+                return remote.executeAsync(function(done) {
+                    var element = document.getElementById('main-q');
```
Comment (darkwing) [inline tests/ui/tests/header.js]: I know I'm repeating a string we have stored here but this exact JS is being injected into the page so we have to repeat it :/

**Comment by openjck** [comment]
Seeing 6 test failures.

```
=============================== Coverage summary ===============================
Statements   : 96% ( 24/25 )
Branches     : 75% ( 9/12 )
Functions    : 83.33% ( 5/6 )
Lines        : 95.65% ( 22/23 )
================================================================================
firefox 37.0.1 on MAC: 6/25 tests failed

----------------------|-----------|-----------|-----------|-----------|
File                  |   % Stmts |% Branches |   % Funcs |   % Lines |
----------------------|-----------|-----------|-----------|-----------|
   ui/                |        96 |        75 |     83.33 |     95.65 |
      _base.js        |       100 |       100 |       100 |       100 |
      _cli.js         |     94.74 |        75 |        75 |     94.12 |
      _tests.js       |       100 |       100 |       100 |       100 |
      intern-local.js |       100 |       100 |       100 |       100 |
   ui/tests/          |     70.73 |       100 |     67.07 |     70.73 |
      auth.js         |     71.43 |       100 |     69.57 |     71.43 |
      demos.js        |       100 |       100 |       100 |       100 |
      env.js          |       100 |       100 |       100 |       100 |
      footer.js       |       100 |       100 |       100 |       100 |
      header.js       |     69.23 |       100 |     66.67 |     69.23 |
      homepage.js     |     84.21 |       100 |     84.62 |     84.21 |
      wiki.js         |     28.57 |       100 |     23.53 |     28.57 |
   ui/tests/lib/      |     84.42 |     67.86 |     88.24 |     84.21 |
      POM.js          |     85.19 |     66.67 |        80 |     84.62 |
      assert.js       |       100 |       100 |       100 |       100 |
      config.js       |       100 |       100 |       100 |       100 |
      login.js        |     69.57 |        25 |     83.33 |     69.57 |
      poll.js         |     93.33 |      87.5 |       100 |     93.33 |
----------------------|-----------|-----------|-----------|-----------|
All files             |     78.22 |        70 |     73.77 |     77.93 |
----------------------|-----------|-----------|-----------|-----------|

TOTAL: tested 1 platforms, 6/25 tests failed (5 skipped)
```

**Comment by darkwing** [comment]
lol, need details dogg

**Comment by openjck** [comment]
All output:

https://pastebin.mozilla.org/8830073

**Comment by darkwing** [comment]
Derp.  Let's use staging as our testing base so we're working off of the same content, waffle flags, etc.

**Pull request merged**